test/units/TEST-23-UNIT-FILE.ExtraFileDescriptors.sh

```diff
@@ -51,6 +51,17 @@ timeout 10s bash -xec 'while [[ "$(systemctl show -P SubState test-23-extra-fd.s
 assert_eq "$(systemctl show -P Result "$TEST_UNIT")" "success"
 assert_eq "$(systemctl show -P ExecMainStatus "$TEST_UNIT")" "0"
 
+# Verify extra file descriptors stay accessible even after service manager re-executes
+systemctl daemon-reexec
+
+systemctl start "$TEST_UNIT"
```

Comment:
Hmm, IIUC this should be restart? start on exited service is no-op?

Reply:
Ah yes. And now I'm getting failures that the service is not finding the FD...debugging

Reply:
I suppose you need to reset fd read offset after restarting.

Reply:
I.e. maybe replace read -u in openfile-child.sh with cat /proc/self/fd/...?

Reply:
Ohhhhhhh yea the file pointer is at the end of the file. duh. Let me add `seek -u $fd 0` and see if that works, otherwise yea cat/ proc/self/fd will work swimmingly

Reply:
Fixed in latest version. It affects OpenFile test too...so I can make another (similar) script if we want to validate read offset in OpenFile is 0...up to you!

Reply:
ugh, openfile relies on read escaping some backslashes. For simplicity, I will make a new script specifically for this test